api/pkg/imagebuilder/imagebuilder.go

```diff
@@ -474,24 +473,48 @@ func (c *imageBuilder) createKanikoJobSpec(project mlp.Project, model *models.Mo
 		}
 	}
 
-	resourceRequirements := RequestLimitResources{
-		Request: Resource{
-			CPU: resource.MustParse(
-				c.config.Resources.Requests.CPU,
-			),
-			Memory: resource.MustParse(
-				c.config.Resources.Requests.Memory,
-			),
-		},
-		Limit: Resource{
-			CPU: resource.MustParse(
-				c.config.Resources.Limits.CPU,
-			),
-			Memory: resource.MustParse(
-				c.config.Resources.Limits.Memory,
-			),
-		},
-	}
+	if resource != nil {
```

Comment:
There are four values here CPU/Mem request+limit. The end behaviour is as long as user provide either CPU/Mem request override, it will be use, for the one not passed, it will use default.



We can keep the original block then check if `if resource != nil` (can we rename to imageBuilderResource or imageBuilderResourceRequest? resource is too vague)

for each field if the imageBuilderResourceRequest is not null, override the default. 

The last part to watch is if the request > default limits. In this case we can set limits==user requested resource. This part will fail at k8 creation if request > limit.